lib/Models/Cesium.ts

```diff
@@ -0,0 +1,1221 @@
+                    var camera = that.scene.camera;
+
+                    // Work out the destination that the camera would naturally fly to
+                    var destinationCartesian = camera.getRectangleCameraCoordinates(
+                        viewOrExtent
+                    );
+                    var destination = Ellipsoid.WGS84.cartesianToCartographic(
+                        destinationCartesian
+                    );
+                    var terrainProvider = that.scene.globe.terrainProvider;
+                    var level = 6; // A sufficiently coarse tile level that still has approximately accurate height
+                    var positions = [Rectangle.center(viewOrExtent)];
+
+                    // Perform an elevation query at the centre of the rectangle
+                    return sampleTerrain(
+                        terrainProvider,
+                        level,
+                        positions
+                    ).then(function(results) {
+                        var finalDestinationCartographic = {
+                            longitude: destination.longitude,
+                            latitude: destination.latitude,
+                            height: destination.height + results[0].height
+                        };
+
+                        var finalDestination = Ellipsoid.WGS84.cartographicToCartesian(
+                            finalDestinationCartographic
+                        );
+
+                        camera.flyTo({
+                            duration: flightDurationSeconds,
+                            destination: finalDestination
+                        });
+                    });
+                } else if (viewOrExtent.position !== undefined) {
+                    that.scene.camera.flyTo({
+                        duration: flightDurationSeconds,
+                        destination: viewOrExtent.position,
+                        orientation: {
+                            direction: viewOrExtent.direction,
+                            up: viewOrExtent.up
+                        }
+                    });
+                } else {
+                    that.scene.camera.flyTo({
+                        duration: flightDurationSeconds,
+                        destination: viewOrExtent.rectangle
+                    });
+                }
+            })
+            .then(function() {
+                // that.notifyRepaintRequired();
+            });
+    }
+}
+
+const createImageryLayer: (ip: Cesium.ImageryProvider) => Cesium.ImageryLayer = createTransformer((ip: Cesium.ImageryProvider) => {
+    console.log('Creating a new ImageryLayer');
+    return new ImageryLayer(ip);
+});
+
+function makeImageryLayerFromParts(parts: ImageryParts): Cesium.ImageryLayer {
+    const layer = createImageryLayer(parts.imageryProvider);
+
+    // This should only be called on ImageryParts that are observable. I think we can assume
+    autorun(() => {
```

Comment:
I don't think this autorun is needed; we can just assign the properties once here. If the alpha changes, this function will automatically be invoked again and update the Cesium layer's `alpha` property.

Reply:
If I'm right, that's a really good thing because currently this autorun is a leak.

Reply:
That'll run the entire map items <-> Cesium ImageryLayerCollection every time you change opacity

Reply:
As discussed on Slack, yeah, that mapping re-runs every time, but it's not as expensive as it sounds, and it runs every time even with this autorun in place. Might be room for improvement, but I think it's pretty much fine as is.